M2/Macaulay2/e/NCAlgebras/SuffixTree.cpp

```diff
@@ -0,0 +1,713 @@
+#include "NCAlgebras/SuffixTree.hpp"
+#include <algorithm>            // for copy
+#include <iterator>             // for back_insert_iterator, back_inserter
+#include "NCAlgebras/Word.hpp"  // for Word, operator<<
```

Comment:
This comment applies to almost all the new sources, but I didn't notice it till now:
I think we should stick to the [convention here](https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes) for order of inclusions. This file seems to be correct (except newlines), but all the others may need to be reordered. It's not hugely urgent though.